test/models/_model_test_case.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from abc import abstractmethod
+
+
+class ModelTestCase(object):
+    @abstractmethod
+    def create_model(self, train_data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def create_test_data(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def create_batch_test_data(self):
+        raise NotImplementedError()
+
+    def test_forward_train(self):
+        data = self.create_test_data()
+        model = self.create_model(data)
+        model.train()
+        output = model(data)
+        self.assertTrue(output.lazy_covariance_matrix.dim() == 2)
+        self.assertTrue(output.lazy_covariance_matrix.size(-1) == data.size(-2))
+        self.assertTrue(output.lazy_covariance_matrix.size(-2) == data.size(-2))
```

Comment:
We should probably also add checks here about the gradients of the hyperparameters not being zero.